src/app/content/highlights/selectors.ts

```diff
@@ -32,6 +32,17 @@ export const highlightLocationFilters = createSelector(
   : new Map() as HighlightLocationFilters
 );
 
+export const totalCountsPerPage = createSelector(
+  localState,
+  (state) => state.summary.totalCountsPerPage
+);
+
+export const totalCountsPerLocation = createSelector(
+  localState,
```

Comment:
because you're depending on `localState` here the selector will be re-evaluated anytime the highlight state changes at all, if you put `totalCountsPerPage` here, it'll only re-evaluate when that value changes